packages/camera/ios/Classes/CameraPlugin.m

```diff
@@ -68,14 +79,46 @@ - (void)captureOutput:(AVCapturePhotoOutput *)output
   NSData *data = [AVCapturePhotoOutput
       JPEGPhotoDataRepresentationForJPEGSampleBuffer:photoSampleBuffer
                             previewPhotoSampleBuffer:previewPhotoSampleBuffer];
+  UIImage *image = [UIImage imageWithCGImage:[UIImage imageWithData:data].CGImage
+                                       scale:1.0
+                                 orientation:[self getImageRotation]];
   // TODO(sigurdm): Consider writing file asynchronously.
-  bool success = [data writeToFile:_path atomically:YES];
+  bool success = [UIImageJPEGRepresentation(image, 1.0) writeToFile:_path atomically:YES];
   if (!success) {
     _result([FlutterError errorWithCode:@"IOError" message:@"Unable to write file" details:nil]);
     return;
   }
   _result(nil);
 }
+
+- (UIImageOrientation)getImageRotation {
```

Comment:
Would it be possible to use `UIDevice.currentDevice.orientation` to get device orientation instead of using the accelerometer?

Reply:
Never mind. I just noticed the number 3 was referring to this.